File: JavaScript/pangram.js
//check if given string is pangram or not

function checkPangram(givenString){
  givenString =  givenString.toLowerCase().replaceAll(" ",'');
  const count = new Set(givenString);
  if(count.size === 26) console.log("Given String is Pangram");
  else console.log("Given String is not Pangram");

}
givenString = "The quick brown fox jumps over the lazy dog"
checkPangram(givenString);

/*
Sample Test case 1:
    Input: 'abc'
    Output: Given String is not pangram

Sample Test case 2:
    Input: 'The quick brown fox jumps over the lazy dog'
    Output: Given String is pangram

Time Complexity : O(n)
Space Complexity : O(n)
*/

File: C-Plus-Plus/Pangram.cpp
#include <bits/stdc++.h>
using namespace std;
bool isPangram(string str)
{
    //converting the string to lowercase
    transform(str.begin(), str.end(), str.begin(), ::tolower);
    int n = str.length();
    int count[26] = {0};
    for (int i = 0; i < n; i++)
    {
        if (str[i] == ' ')
        {
            continue; // omitting space
        }
        count[str[i] - 'a']++;
    }
    for (int i = 0; i < 26; i++)
    {
        if (count[i] == 0)
        {
            return false;
        }
    }
    return true;
}
int main()
{
    string str;
    getline(cin, str);
    if (isPangram(str))
    {
        cout << "String is a pangram\n";
    }
    else
    {
        cout << "String is not a pangram\n";
    }
}
/*
Time Complexity: O(n)
Space Complexity: O(1)

Input: The quick brown fox jumps over the lazy dog
Output : String is a pangram

*/
